contrib_bots/lib/gcalendar.py

```diff
@@ -0,0 +1,331 @@
+
+def send_parsing_error_message(err, message, client):
+    error = ''
+
+    if err.error_id == 1:
+        error = ('Unknown message format.\n\n'
+                 'Usage examples:\n'
+                 '```\n'
+                 '@gcalendar Meeting with John | 2017/03/14 13:37 | 2017/03/14 15:00:01 | GMT\n'
+                 '@gcalendar Trip to LA | tomorrow\n'
+                 '```\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 2:
+        error = ('Unknown date format.\n\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+    elif err.error_id == 3:
+        error = ('Unknown timezone.\n\n'
+                 'Please, use a numerical (`+01:00`) or abbreviated (`CET`) timezone.\n'
+                 'Send `@gcalendar help` for detailed usage instructions.')
+
+    reply_message(error, message, client)
+
+class GCalendarHandler(object):
+    '''
+    This plugin facilitates creating Google Calendar events.
+
+    Usage:
+        For delimited events:
+            @gcalendar <event_title> | <start_date> | <end_date> | <timezone> (optional)
+        For full-day events:
+            @gcalendar <event_title> | <start_date>
+
+    The "event-title" supports all characters but pipes (|)
+
+    Timezones can be specified in both numerical (+00:00) or abbreviated
+    format (UTC).
+    If no timezone is specified, the server's will be considered as default.
+
+    Right now it only works for the calendar set as "primary" in your account.
+
+    Please, run the script api/integrations/google/oauth.py before using this bot,
+    in order to provide it the necessary access to your Google account.
+    '''
+    def __init__(self):
+        try:
+            store = Storage(CREDENTIAL_PATH)
+            credentials = store.get()
+        except IOError:
+            logging.error('Couldn\'t find valid credentials.\n'
+                          'Run the oauth.py script in'
+                          'api/integrations/google/oauth.py first.')
+            sys.exit(1)
+
+        http = credentials.authorize(httplib2.Http())
+        self.service = discovery.build('calendar', 'v3', http=http)
+
+    def usage(self):
+        return '''
+            This plugin will allow users to create events
+            in a Google Calendar. Users should preface
+            messages with "@gcalendar".
+
+            Before running this, make sure to register a new
+            project in the Google Developers Console
+            (https://console.developers.google.com/start/api?id=calendar),
+            download "client secret" as a JSON file, and run
+            api/integrations/google/oauth.py for the Calendar
+            read and write scope
+            (https://www.googleapis.com/auth/calendar).
+            '''
+
+    def triage_message(self, message, client):
+        original_content = message['content']
+
+        is_gcalendar = original_content.startswith('@gcalendar')
```

Comment:
I think we can just move it to return statement - defining `is_gcalendar` is not needed.